code/src/item_effect.c

```diff
@@ -13,7 +13,10 @@ void ItemEffect_None(SaveContext* saveCtx, s16 arg1, s16 arg2) {
 }
 
 void ItemEffect_FullHeal(SaveContext* saveCtx, s16 arg1, s16 arg2) {
-    saveCtx->healthAccumulator = 20 * 0x10;
+    //With the No Health Refills option on, store-bought health upgrades do not heal the player
+    if(!gSettingsContext.heartDropRefill == HEARTDROPREFILL_NOREFILL && !gSettingsContext.heartDropRefill == HEARTDROPREFILL_NODROPREFILL){

```

Comment:
I think that there is errors in this condition due to missing parenthesis. You probably want it to be like
```
if((gSettingsContext.heartDropRefill != HEARTDROPREFILL_NOREFILL) && (gSettingsContext.heartDropRefill != HEARTDROPREFILL_NODROPREFILL))
```